doc/source/text.rst

```diff
@@ -205,6 +217,52 @@ and optional groups like
 
 can also be used.
 
+When ``str.extract`` method is used on an ``Index``, it returns a
+``DataFrame`` with as many columns as capture groups.
+
+.. ipython:: python
+
+   s = pd.Series(["a1", "b2", "c3"], ["A11", "B22", "C33"])
+   s
+   s.index.str.extract("(?P<letter>[a-zA-Z])")
+   s.index.str.extract("(?P<letter>[a-zA-Z])(?P<numbers>[0-9]+)")
+
+.. _text.extractall:
+
+Unlike ``extract`` (which returns only the first match),
+
+.. ipython:: python
+
+   s = pd.Series(["a1a2", "b1", "c1"], ["A", "B", "C"])
+   s
+   s.str.extract("[ab](?P<digit>\d)")
+
+.. versionadded:: 0.18.0
+
+the ``extractall`` method (introduced in version 0.18) returns a
```

Comment:
you don't need to say introduced in 0.18, that's what the ``versionadded`` tag does